sql/hive/src/test/scala/org/apache/spark/sql/hive/client/VersionsSuite.scala

```diff
@@ -584,10 +586,11 @@ class VersionsSuite extends SparkFunSuite with Logging {
 
     test(s"$version: sql read hive materialized view") {
       // HIVE-14249 Since Hive 2.3.0, materialized view is supported.
-      // But skip Hive 3.1 because of SPARK-27074.
-      if (version == "2.3") {
+      if (version == "2.3" || version == "3.1") {
+        val disableRewrite = if (version == "2.3") "" else "DISABLE REWRITE"
```

Comment:
Could you add a comment about the reason why we use `DISABLE REWRITE` in `Hive 3.x`?

Reply:
OK.
https://github.com/apache/hive/blob/8fd90ea13e24c2de5e4abbd9e6ea427ede559a6c/ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java#L13456-L13460